cmd/controller/controller.go

```diff
@@ -168,12 +198,23 @@ func (onosClient *OnosService) GetRogueIPs(rogueIPChannel chan RogueIPs) {
 		}
 
 		var rogueIPs RogueIPs
-		if rsp != nil && rsp.Body != nil {
-			json.NewDecoder(rsp.Body).Decode(&rogueIPs)
-			logger.ControllerLog.Infoln("received rogueIPs from Onos App: ", rogueIPs)
-			//writing rogueIPs into channel
-			rogueIPChannel <- rogueIPs
+		if rsp != nil {
+			if rsp.Body != nil {
+				err := json.NewDecoder(rsp.Body).Decode(&rogueIPs)
+				if err != nil {
+					logger.ControllerLog.Errorln("OnosApp response body decode failed: ", err)
+				} else {
+					logger.ControllerLog.Infoln("received rogueIPs from Onos App: ", rogueIPs)
+					if validateIPs(rogueIPs) == nil {
+						//writing rogueIPs into channel
+						rogueIPChannel <- rogueIPs
+					}
+				}
+			} else {
+				logger.ControllerLog.Infoln("Http Response Body from OnosApp is empty")
```

Comment:
Check with Sunder that what shall be the response code/body incase no rogue IP found.